utils/src/main/java/software/amazon/awssdk/utils/http/SdkHttpUtils.java

```diff
@@ -322,4 +328,15 @@ public static String appendUri(String baseUri, String path) {
     public static boolean isSingleHeader(String h) {
         return SINGLE_HEADERS.contains(StringUtils.lowerCase(h));
     }
+
+    /**
+     * Extracts query parameters from the given URI
+     */
+    public static Map<String, List<String>> uriParams(URI uri) {
+        return Pattern.compile(QUERY_PARAM_PATTERN)
+               .splitAsStream(uri.getQuery().trim())
```

Comment:
BTW, I think we should be looking at `getRawQuery()` here, since `getQuery()` will give you the decoded form which is not what we want

Reply:
You're right, updated and adapted unit test